src/view/screens/Moderation.tsx

```diff
@@ -109,10 +121,112 @@ export function ModerationScreen({}: Props) {
           <Trans>Blocked accounts</Trans>
         </Text>
       </Link>
+      <Text
+        type="xl-bold"
+        style={[
+          pal.text,
+          {
+            paddingHorizontal: 18,
+            paddingTop: 18,
+            paddingBottom: 6,
+          },
+        ]}>
+        <Trans>My Account</Trans>
+      </Text>
+      <PwiOptOut />
     </CenteredView>
   )
 }
 
+function PwiOptOut() {
+  const pal = usePalette('default')
+  const {_} = useLingui()
+  const {currentAccount} = useSession()
+  const {data: profile} = useProfileQuery({did: currentAccount?.did})
+  const updateProfile = useProfileUpdateMutation()
+
+  const isOptedOut =
+    profile?.labels?.some(l => l.val === '!no-unauthenticated') || false
+  const canToggle = profile && !updateProfile.isPending
+
+  const onToggleOptOut = React.useCallback(() => {
+    if (!profile) {
+      return
+    }
+    updateProfile.mutate({
+      profile,
+      updates: existing => {
```

Comment:
I realize this is not new, but are we mutating something that's also used for rendering? Generally in React we shouldn't mutate stuff we render — can this be a new object?

Reply:
It's not being used for rendering. I agree with the general approach as a safe default approach, but this is a narrow non-react api that handles this okay. Not against changing it, but not an issue